Classes/FileStorage/FileStorage.php

```diff
@@ -157,4 +159,22 @@ public function createFolder(string $folderIdentifier): void
             }
         }
     }
+
+    public function getTempPath(): string
+    {
+        return Environment::getVarPath() . '/transient/';
+    }
+
+    public function writeTempFile(string $filePrefix = '', string $fileContent = '', string $fileSuffix = ''): string|bool
+    {
+        $result = null;
+        $filePath = GeneralUtility::tempnam($filePrefix, $fileSuffix);
+        if ($this->fileIsWriteable($filePath)) {
```

Comment:
@mkarulin `FileStorage::fileIsWriteable()` of the TYPO3 package expects a file identifier, not a file path. I think this call will always return `true` because it returns the negated result of `FileStorage::fileIsReadOnly()` which will return `false` if the file identifier can't be resolved.

I would consider this a bug in `FileStorage::fileIsWriteable()`, but also, the method should not be used here in the first place.